data-products/src/main/scala/org/sunbird/analytics/audit/ScoreMetricMigrationJob.scala

```diff
@@ -97,10 +116,38 @@ object ScoreMetricMigrationJob extends optional.Application with IJob with BaseR
     aggregation.toList.flatten.toMap
   }
 
-  def getBestScoreRecordsDF(assessmentDF: DataFrame): DataFrame = {
-    val df = Window.partitionBy("user_id", "batch_id", "course_id", "content_id").orderBy(desc("total_score"))
-    assessmentDF.withColumn("rownum", row_number.over(df)).where(col("rownum") === 1).drop("rownum")
-      .withColumn("batchid", concat(lit("cb:"), col("batch_id"))).drop("batch_id")
+  def mergeAggDetailListValues(): Seq[Seq[String]] => List[String] = (aggregation: Seq[Seq[String]]) => {
+    aggregation.toList.flatten
+  }
+
+  def mergeAggDetailsCol(): (Seq[String], List[String]) => List[String] = (aggDetails: Seq[String], newAggDetails: Seq[String]) => {
```

Comment:
Right now this is a one-time script and we will not be having any records in the new column so it better will have conditions or check to merge the agg details. will read the condition value from the config